### DISCUSSION TITLE
Cant find my site on google

### DISCUSSION THREAD

**Discussion by webface** [vercel/next.js#11705] [Help] [answered]
I created a nextjs site. It has a sitemap all of which I can see in my google console. when i go to site:mysite.com I can see all my pages indexed but when I do a search for my site on google, I cant find it. I have meta keyword and meta descriptions and Ive verified my domain with google. Any help?

**Comment by c0b41** [comment]
Url?

**Reply by webface** [reply]
www.tommyadeniyi.com

**Reply by c0b41** [reply]
I don't see any meta tag org structured data

**Reply by webface** [reply]
theres meta keywords and meta descriptions

**Reply by c0b41** [reply]
you are right sorry, maybe your content not ranking?

**Comment by timneutkens** [comment, marked as answer]
Seems that the site is indexed just fine 🤔 

<img width="858" alt="Screen Shot 2020-04-07 at 10 47 12" src="https://user-images.githubusercontent.com/6324199/78649226-72e53f00-78bd-11ea-8842-54c69792960f.png">

**Comment by webface** [comment]
@Timer  @timneutkens I dunno how this is the answer but ive built 3 sites with nextjs and while they are indexed, I cant find any of them on google.
```
// next.config.js
require('dotenv').config()
const prod = process.env.NODE_ENV === 'production'
const withPlugins = require('next-compose-plugins');
const withTM = require('next-transpile-modules')(['@mui-treasury/layout', '@mui-treasury/mockup']);

module.exports = withPlugins([withTM], {
  env:{
    'API_URL':prod ? 'https://api.tommyadeniyi.com' : 'http://localhost:3018',
    'SERVER_URL':prod ? 'http://node-be:3018' : 'http://localhost:3018'
  }
 });
```
In this case, I'm running in a docker container but I have a usecase that is just a shell startup and that site cant be found on google either.
I even have a GA tracking code
Should I be doing something different?

**Reply by Timer** [reply]
I can find you fine on Google. Appearing in search results is all about how you rank, so this question belongs on SEO communities for them to give you tips. The lack of your ranking has nothing to do with Next.js.

![](https://i.imgur.com/PcWt3HB.gif)

**Comment by matttearth** [comment]
Google is cautious with new domains and it also likes domains which have a lot of external backlinks from quality/relevant websites. Your domain is very new (63 days old) and it appears you may be lacking external backlinks. 

To improve your results in Google, try and gain more backlinks from other websites. There are many SEO related activities you could pursue. Since you're a developer adding structured data would be a good option (you have a dev advantage over many websites owners which either lack the knowledge, or combination of foresight and ability to implement structured data). It's not hard, and used correctly structured data can give a website a great advantage.

Moz have some great free resources for learning SEO 
https://moz.com/learn/seo
https://moz.com/blog/structured-data-for-seo-1

I think you've done well to get the site indexed. I've been looking at a load of Nextjs sites and seen many which haven't been indexed so well. I'm exploring this forum to understand why.... We're considering Nextjs for a new solution which has to perform well in organic search.

**Comment by adjieindrawan** [comment]
Hi @timneutkens , currently my site cannot be found in google, i tried search `site: kliniksehat.id` only my subdomain indexed. Please help me.
My domain is https://kliniksehat.id/. Thank you

**Reply by timneutkens** [reply]
Seems to be indexed fine: https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fkliniksehat.id%2F&oq=site%3Ahttps%3A%2F%2Fkliniksehat.id%2F&gs_lcrp=EgZjaHJvbWUyBggAEEUYOTIGCAEQRRg60gEIMjExOGowajeoAgCwAgA&sourceid=chrome&ie=UTF-8

**Reply by c0b41** [reply]
Only Preview Deployments not gonna work, other than will work fine https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines
